pkg/mynewt-core/patches/0001-riot-add-interested-header-symlinks.patch

```diff
@@ -0,0 +1,1930 @@
+From eadc071053d2c929f88b2721d57f49ec351fc035 Mon Sep 17 00:00:00 2001
+From: Francisco Molina <femolina@uc.cl>
+Date: Mon, 19 Apr 2021 15:48:50 +0200
+Subject: [PATCH 1/4] riot: add interested header symlinks
+
+---
+ riot/include/hal/hal_timer.h   |   1 +
```

Comment:
do we really need to add these headers via patch? Can't we just put them into their own include dir(s) in `pkg/mynewt-core/...` and use them from there?

Reply:
I tried to do a symlink so that the files would stay in sync if ever changed... but it ended up being a huge patch, and I did this to avoid unwanted headers being included. What I wanted to avoid was cluttering the code base with files that are exact copies of files in `mynewt-core`. I'll see if there is another solution I can try.

Reply:
I have the following options:

1. one patch that similink all interested headers it would look like https://github.com/apache/mynewt-core/commit/8f03dc5f753986d815ef8cea273b19f62f8b7c78
1. add headers under `pkg/mynewt-core/...`
1. add directories to `INCLUDES` with `CFLAGS+=-DHEADER_GUARD` for all headers that we do not want to include, also if there is a name conflict our header must be included first.

From a maintenance perspective I prefer 1 since it requires the fewest modifications at the price of a very simple patch. Next I would prefer 3 but it means adding 20+ CFLAGS and hoping the `INCLUDES` order never gets changed. Do you see other cons with 1? (note that in the current patch I seem to have copied instead of symlink most of those files)

Reply:
@kaspar030 suggested that I should just use my fork as the reference so to not have to go through patches. That is also an option.

Reply:
1. and 3. I really don't like: symlinks are IMO quite error prone, and everything depending on the order of include paths will (at least with our current build system) explode at some point and will be quite nasty to debug...
2. although this could easily lead to a situation where the copied header get out of sync with their upstream versions, this seems to be slightly cleaner to me...

A fork seems to be a nice option to me. The benefit I see, is that this would make it much easier for us to have a plain view on the files actually used. But this would still leave the problem in keeping the fork and the upstream mynewt repo in sync...

Aside from those options, I think I have another one we could consider: How about simply deleting the headers we are not using from the include dir? This would be a 2 liner in the Makefile and if we put the list of files to delete in a simple variable these two lines would also very easy to maintain, right?!

Reply:
Did as you suggested!